lib/modules/dosomething/dosomething_northstar/dosomething_northstar.auth.inc

```diff
@@ -84,18 +84,35 @@ function dosomething_northstar_openid_connect_claims_alter(&$claims) {
  */
 function dosomething_northstar_openid_connect_post_authorize($tokens, $account, $userinfo, $provider) {
   $edit = [];
+
   dosomething_user_set_fields($edit, [
     'access_token' => $tokens['access_token'],
     'refresh_token' => $tokens['access_token'],
     'access_token_expiration' => $tokens['expire'],
   ]);
 
+  // Make sure the user's Northstar ID is set in the field_data_field_northstar_id table.
+  if (!dosomething_user_get_northstar_id($account->uid)) {
+    dosomething_northstar_save_id_field($account->uid, ['data' => $userinfo]);
+  }
+
   // If we saved a "post-authorization" action, do it now.
   dosomething_northstar_perform_authorization_actions();
 
   // Let's remember that this is a OpenID Connect session.
   $_SESSION['DOSOMETHING_NORTHSTAR_OPENID_CONNECT'] = true;
 
+  if (empty($account->language)) {
+    // Determine the user's language code if possible.
+    $language_code = dosomething_settings_get_geo_country_code();
```

Comment:
I can give that a whirl... wasn't sure since the user isn't anonymous, just doesn't have a `language` set and it's coming from a redirect from Northstar.